lib/resource-timing.js

```diff
@@ -0,0 +1,228 @@
+
+  var contains = function(array, value) {
+      if(Array.prototype.indexOf) {
+        return array.indexOf(value) != -1;
+      } else {
+        var i, length = array.length;
+        for(i = 0; i < length; i++) {
+          if(array[i] === value) {
+            return true;
+          }
+        }
+        return false;
+      }
+    };
+
+  /**
+   * Frontend Web Performance Data Gathering
+   *
+   * @class SurfNPerf
+   */
+  var SurfNPerfRT = function() {
+      this._resourceTiming = null;
+
+      this.initialize();
+    },
+    SNPRTProto = SurfNPerfRT.prototype;
+
+  SNPRTProto.initialize = function() {
+    this._setResourceTimingApi();
+  };
+
+  SNPRTProto._setResourceTimingApi = function() {
+    if(window.performance) {
+      this._resourceTiming = !!(window.performance.getEntriesByType);
+    } else {
+      this._resourceTiming = false;
+    }
+  };
+
+  SNPRTProto._inList = function(origin, options) {
+    options = options || {};
+    if (options.hasOwnProperty("whitelist")) {
+      return contains(options.whitelist, origin);
+    } else if (options.hasOwnProperty("blacklist")) {
+      return !contains(options.blacklist, origin);
+    }
+    else { // No Options
+      return true;
+    }
+  }
+
+  SNPRTProto._getURLOrigin = function(url) {
+    var a = document.createElement('a');
+    a.href = url;
+    return a.protocol + "//" + a.hostname + (a.port ? ':' + a.port : '');
+
+  };
+
+  SNPRTProto.getOrigins = function(options) {
+    if(this._resourceTiming) {
+      var resources = window.performance.getEntriesByType("resource");
+      var origins = [];
+      for(var resource in resources) {
+        var URLOrigin = this._getURLOrigin(resources[resource].name);
+        if(!contains(origins, URLOrigin) && this._inList(URLOrigin, options)) {
+          origins.push(URLOrigin);
+        }
+      }
+      return origins;
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto.getResourcesFromOrigin = function(origin) {
+    if(this._resourceTiming) {
+      var resources = window.performance.getEntriesByType("resource");
+      var resourcesFromOrigin = [];
+      for(var resource in resources) {
+        var URLOrigin = this._getURLOrigin(resources[resource].name);
+        if (URLOrigin === origin) {
+          resourcesFromOrigin.push(resources[resource]);
+        }
+      }
+      return resourcesFromOrigin;
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto.isBufferFull = function() {
+    if(this._resourceTiming) {
+      return !!SURF_N_PERF.resourcetimingbufferfull;
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto._name = function(name) {
+    if(name.charAt(0) == '/') {
+      return window.location.protocol + "//" + window.location.host + name;;
+    } else {
+      return name;
+    }
+  };
+
+  SNPRTProto.getResource = function(name) {
+    if(this._resourceTiming) {
+      name = this._name(name);
+      return window.performance.getEntriesByName(name, "resource")[0]
```

Comment:
no need to set `name = this._name(name);`, just call `getEntriesByName` with `(this._name(name), "resource")[0]`